File: Daybreak.GWCA/source/payloads/UserPayload.cpp
#include <cstdint>
#include <json.hpp>
#include <payloads/UserPayload.h>

using json = nlohmann::json;

namespace Daybreak {
    void to_json(json& j, const UserPayload& p) {
        j = json
        {
            {"Email", p.Email},
            {"CurrentKurzickPoints", p.CurrentKurzickPoints},
            {"CurrentLuxonPoints", p.CurrentLuxonPoints},
            {"CurrentImperialPoints", p.CurrentImperialPoints},
            {"CurrentBalthazarPoints", p.CurrentBalthazarPoints},
            {"CurrentSkillPoints", p.CurrentSkillPoints},
            {"TotalKurzickPoints", p.TotalKurzickPoints},
            {"TotalLuxonPoints", p.TotalLuxonPoints},
            {"TotalImperialPoints", p.TotalImperialPoints},
            {"TotalBalthazarPoints", p.TotalBalthazarPoints},
            {"TotalSkillPoints", p.TotalSkillPoints},
            {"MaxKurzickPoints", p.MaxKurzickPoints},
            {"MaxLuxonPoints", p.MaxLuxonPoints},
            {"MaxImperialPoints", p.MaxImperialPoints},
            {"MaxBalthazarPoints", p.MaxBalthazarPoints},
        };
    }
}

File: Daybreak/Services/Scanner/Models/UserPayload.cs
namespace Daybreak.Services.Scanner.Models;

internal sealed class UserPayload
{
    public string? Email { get; set; }
    public uint CurrentKurzickPoints { get; set; }
    public uint CurrentLuxonPoints { get; set; }
    public uint CurrentImperialPoints { get; set; }
    public uint CurrentBalthazarPoints { get; set; }
    public uint CurrentSkillPoints { get; set; }
    public uint TotalKurzickPoints { get; set; }
    public uint TotalLuxonPoints { get; set; }
    public uint TotalImperialPoints { get; set; }
    public uint TotalBalthazarPoints { get; set; }
    public uint TotalSkillPoints { get; set; }
    public uint MaxKurzickPoints { get; set; }
    public uint MaxLuxonPoints { get; set; }
    public uint MaxImperialPoints { get; set; }
    public uint MaxBalthazarPoints { get; set; }
}
